packages/sitecore-jss/src/graphql/search-service.ts

```diff
@@ -1,5 +1,19 @@
-import { GraphQLClient } from './../graphql-request-client';
 import { DocumentNode } from 'graphql';
+import { GraphQLClient } from '../graphql-request-client';
+
+/**
+ * Data needed to paginate the site results
+ */
+export interface PageInfo {
```

Comment:
I feel like we could take this even one step further and include the `results` and `total` the next level up in all queries with paginated results. Something like this should work?

```
export type PageInfo = {
  endCursor: string;
  hasNext: boolean;
};

export type PaginatedResult<T> = {
  pageInfo: PageInfo;
  results: T[];
  total: number;
};

export type SearchQueryResult<T> = {
  search: PaginatedResult<T>;
};

export type SiteRouteQueryResult<T> = {
  site: {
    siteInfo: {
      routes: PaginatedResult<T>;
    };
  };
};
```

Reply:
"total" is not used anywhere and the way result objects are setup right now are already simplistic and descriptive enough:
`  site: {
    siteInfo: {
      routes: {
        pageInfo: PageInfo;
        results: T[];`
        
and 
        `search: {
           pageInfo: PageInfo;
           results: T[];`

Reply:
That is to say: I think putting pageInfo and results into another structure will be right in spirit, but would make us lose on readability of the code.

Reply:
Ah, we could drop 'total' then. And totally fine to leave as-is. Just thought it worth mentioning in the sprit of reducing duplicate types.